tools/tpm2_getpubak.c

```diff
@@ -245,27 +239,34 @@ static bool create_ak(TSS2_SYS_CONTEXT *sapi_context) {
 
     memcpy(&sessions_data.auths[0].hmac, &ctx.passwords.endorse, sizeof(ctx.passwords.endorse));
 
-    SESSION *session = NULL;
-    UINT32 rval = tpm_session_start_auth_with_params(sapi_context, &session, TPM2_RH_NULL, 0, TPM2_RH_NULL, 0,
-            &nonce_caller, &encrypted_salt, TPM2_SE_POLICY, &symmetric,
-            TPM2_ALG_SHA256);
-    if (rval != TPM2_RC_SUCCESS) {
-        LOG_ERR("tpm_session_start_auth_with_params Error. TPM Error:0x%x", rval);
+    tpm2_session_data *data = tpm2_session_data_new(TPM2_SE_POLICY);
+    if (!data) {
+        LOG_ERR("oom");
+        return false;
+    }
+
+    tpm2_session *session = tpm2_session_new(sapi_context, data);
+    if (!session) {
+        LOG_ERR("Could not start tpm session");
         return false;
     }
 
     LOG_INFO("tpm_session_start_auth_with_params succ");
 
-    rval = TSS2_RETRY_EXP(Tss2_Sys_PolicySecret(sapi_context, TPM2_RH_ENDORSEMENT,
-            session->sessionHandle, &sessions_data, 0, 0, 0, 0, 0, 0, 0));
+    TPMI_SH_AUTH_SESSION handle = tpm2_session_get_session_handle(session);
+    tpm2_session_free(&session);
+
+
+    TPM2_RC rval = TSS2_RETRY_EXP(Tss2_Sys_PolicySecret(sapi_context, TPM2_RH_ENDORSEMENT,
+            handle, &sessions_data, 0, 0, 0, 0, 0, 0, 0));
```

Comment:
Again I think it's much easier to read if the arguments are aligned with the function opening parentheses.